app.js

```diff
@@ -0,0 +1,105 @@
+const session = require("express-session");
+// Initialize an express application
+const express = require('express');
+
+// Import the express-async-errors module to handle async errors
+require("express-async-errors");
+
+const app = express();
+// Set the view engine to ejs for rendering views
+app.set("view engine", "ejs");
+
+// Use body-parser middleware to parse incoming request bodies
+app.use(require("body-parser").urlencoded({ extended: true }));
+
+require("dotenv").config(); // to load the .env file into the process.env object
+
+const MongoDBStore = require('connect-mongodb-session')(session);
+
+const url = process.env.MONGO_URI;
+
+// Create a new MongoDBStore instance for storing sessions in MongoDB
+const store = new MongoDBStore({    
+    uri: url,// The MongoDB connection URI    
+    collection: "mySessions",// The name of the collection where sessions will be stored
+});
+  
+  // Log any errors that occur with the store
+  store.on("error", function (error) {
+    console.log(error);
+  });
+  
+  // the parameters for the session middleware
+  const sessionParms = {    
+    secret: process.env.SESSION_SECRET,// The secret used to sign the session ID cookie    
+    resave: true,// Forces the session to be saved back to the session store, even if the session was never modified during the request
+    
+    saveUninitialized: true,// Forces a session that is "uninitialized" to be saved to the store
+    
+    store: store,// The store instance where sessions will be stored
+    
+    cookie: { secure: false, sameSite: "strict" },// Options for the session ID cookie
+  };
+  
+  // If the application is in production
+  if (app.get("env") === "production") {    
+    app.set("trust proxy", 1);// Trust the first proxy    
+    sessionParms.cookie.secure = true;// Serve secure cookies
+  }
+  
+  // Use the session middleware with the defined parameters
+    app.use(session(sessionParms));
+  // Add the connect-flash middleware
+    app.use(require("connect-flash")());
+
+    app.get("/secretWord", (req, res) => {
+        if (!req.session.secretWord) {
+          req.session.secretWord = "syzygy";
+        }
```

Comment:
if (!req.session.secretWord) {
  req.session.secretWord = "syzygy";
}

Reply:
Oops, this feedback is wrong! 😑 

Instead, I meant to write this…

Nitpick:
Here's a great example of how using the [Logical OR assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_OR_assignment) can simplify your code!
```diff
- if (!req.session.secretWord) {
-    req.session.secretWord = "syzygy";
- }
+ req.session.secretWord ||= "syzygy";
```